macro-plantuml-macro/src/main/java/org/xwiki/contrib/plantuml/internal/PlantUMLMacro.java

```diff
@@ -178,9 +162,13 @@ private String computeServer(PlantUMLMacroParameters parameters)
         return serverURL;
     }
 
-    private String getImageId(String content)
+    private PlantUMLDiagramFormat computeFormat(PlantUMLMacroParameters parameters)
     {
-        return String.valueOf(content.hashCode());
+        PlantUMLDiagramFormat format = parameters.getFormat();
+        if (format == null) {
+            format = PlantUMLDiagramFormat.fromString(this.configuration.getPlantUMLOutputFormat());
```

Comment:
No need for the `fromString` here if `getPlantUMLOutputFormat` return a `PlantUMLDiagramFormat` object.